File: atcoder/abc390a_12345.hs
-- vicfred
-- https://atcoder.jp/contests/abc390/tasks/abc390_a
-- implementation
import Data.Bool

main :: IO ()
main = getLine >>= putStrLn .bool "No" "Yes"
                            .f
                            .map read
                            .words

f :: [Int] -> Bool
f [1,2,3,5,4] = True
f [1,2,4,3,5] = True
f [1,3,2,4,5] = True
f [2,1,3,4,5] = True
f _ = False

File: atcoder/abc390a_12345.rs
// vicfred
// https://atcoder.jp/contests/abc390/tasks/abc390_a
// implementation
use std::io;

fn main() {
    let mut line = String::new();

    io::stdin().read_line(&mut line).unwrap();

    let mut A: Vec<i64> = line
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    for i in 0..4 {
        // XOR swap
        // https://en.wikipedia.org/wiki/XOR_swap_algorithm
        A[i] = A[i + 1] ^ A[i];
        A[i + 1] = A[i] ^ A[i + 1];
        A[i] = A[i + 1] ^ A[i];
        let mut valid: bool = true;
        // println!("{} {} {} {} {}", A[0], A[1], A[2], A[3], A[4]);
        for j in 0i64..5i64 {
            if A[j as usize] != j + 1 {
                valid = false;
                // println!("failed at {}: {}", j, A[j]);
            }
        }
        if valid {
            println!("Yes");
            return;
        }
        A[i] = A[i + 1] ^ A[i];
        A[i + 1] = A[i] ^ A[i + 1];
        A[i] = A[i + 1] ^ A[i];
    }
    println!("No");
}
